### DISCUSSION TITLE
Any way to install docker plugins on bootstrap?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [basecamp/kamal#372] [Q&A]
I'd love to install the loki plugin so I can log directly to loki https://grafana.com/docs/loki/latest/clients/docker-driver/

I don't see any way in the current configuration to do this, but not sure if there was some lower-level mechanic that could be used to run additional commands on the hosts during bootstrap.

I guess I could always run a command on all servers, then update them to use the new logging driver.

**Comment by dhh** [comment]
I do like the idea of having a bootstrap provisioning script. Could add it as a hook with post-bootstrap.

**Reply by danthegoodman1** [reply]
Yeah, fwiw the Loki driver is a bad idea as it blocks shutdown of containers if it hasn’t pushed logs yet, so if you put in a bad config it locks the container and force kill commands don’t work. I ended up just adding Vector to all the mrsk servers
